File: java/src/main/java/euler/Problem006.java
package euler;

public class Problem006 {

    public long differenceBetweenSumOfSquaresAndSquareOfSum(final int n) {
        long squaresSum = squaresCalculate(n);
        long squareOfSum = squaresOfSumCalculate(n);

        return squareOfSum - squaresSum;
    }

    private long squaresOfSumCalculate(final int n) {
        long sum = 0;
        for (long i = 1; i <= n; i++) {
            sum = sum + i;
        }

        return sum * sum;
    }

    private long squaresCalculate(final int n) {
        long squaresSum = 0;
        for (long i = 1; i <= n; i++) {
            squaresSum = squaresSum + (i * i);
        }

        return squaresSum;
    }
}

File: python/src/euler/problem006.py
def squares_of_sum_calculate(n: int) -> int:
    sum = 0
    for i in range(1, n + 1):
        sum +=i

    return sum ** 2

def squares_calculate(n: int) -> int:
    squares_sum = 0
    for i in range(1, n + 1):
        squares_sum += i ** 2

    return squares_sum

def difference_between_sum_of_squares_and_square_of_sum(n: int) -> int:
    squaresSum = squares_calculate(n)
    squareOfSum = squares_of_sum_calculate(n)

    return squareOfSum - squaresSum